common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -503,6 +514,44 @@ def test_downgrade_enrollment_with_mode(self):
         self.assertTrue(is_active)
         self.assertEqual(course_mode, CourseMode.HONOR)
 
+    def test_deactivate_enrollment(self):
+        """With the right API key, deactivate (i.e., unenroll from) an existing enrollment."""
+        # Create an honor and verified mode for a course. This allows an update.
+        for mode in [CourseMode.HONOR, CourseMode.VERIFIED]:
+            CourseModeFactory.create(
+                course_id=self.course.id,
+                mode_slug=mode,
+                mode_display_name=mode,
+            )
+
+        # Create a 'verified' enrollment
+        self.assert_enrollment_status(as_server=True, mode=CourseMode.VERIFIED)
+
+        # Check that the enrollment is 'verified' and active.
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course.id))
+        course_mode, is_active = CourseEnrollment.enrollment_mode_for_user(self.user, self.course.id)
+        self.assertTrue(is_active)
+        self.assertEqual(course_mode, CourseMode.VERIFIED)
+
+        # Verify that a non-Boolean enrollment status is treated as invalid.
+        self.assert_enrollment_status(
+            as_server=True,
+            mode=None,
+            is_active='foo',
+            expected_status=status.HTTP_400_BAD_REQUEST
+        )
+
+        # Verify that the enrollment has been deactivated, and that the mode is unchanged.
+        self.assert_enrollment_status(
+            as_server=True,
+            mode=None,
+            is_active=False,
+            expected_status=status.HTTP_200_OK
+        )
+        course_mode, is_active = CourseEnrollment.enrollment_mode_for_user(self.user, self.course.id)
+        self.assertFalse(is_active)
+        self.assertEqual(course_mode, CourseMode.VERIFIED)
+
```

Comment:
[Nothing](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/models.py#L880); `update_enrollment` is idempotent in that regard.